<h2 style="color:blue;">Name: Sohail Aslam Bhatti</h2>
<h2 style="color:blue;">Roll No. 465885</h2>
<h2 style="color:blue;">Assignment_05: Loops</h2>

<h4 style="color:green;">Q1. Write a Python program to print the number from 1 to 10 using a for loop.

In [1]:
for i in range(1, 11):  
    print(i)

1
2
3
4
5
6
7
8
9
10


<h4 style="color:green;">Q2. Write a Python program to print the number from 20 to 1 using a while loop.

In [2]:
i = 20
while i >= 1:
    print(i)
    i -= 1   

20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1


<h4 style="color:green;">Q3. Write a program to print even numbers from 1 to 10.

In [3]:
for i in range(1, 11):
    if i % 2 == 0:  
        print(i)

2
4
6
8
10


<h4 style="color:green;">Q4. Write a program that prompts the user to enter a number n and prints all the numbers from 1 to n.

In [4]:
num = int(input("Enter a number: "))

for i in range(1, num + 1):
    print(i)

Enter a number:  15


1
2
3
4
5
6
7
8
9
10
11
12
13
14
15


<h4 style="color:green;">Q5. Write a program that prompts the user to enter a number n, and then prints all the odd numbers between 1 and n.

In [5]:
num = int(input("Enter a number: "))

print(f"Odd numbers between 1 and {num}:")
for i in range(1, num + 1):
    if i % 2 != 0:   
        print(i)

Enter a number:  15


Odd numbers between 1 and 15:
1
3
5
7
9
11
13
15


<h4 style="color:green;">Q6. Write a program that prints 'Happy Birthday!' five times on screen.

<h4 style="color:green;">Q7. Write a program that takes a number n as input from the user and generates the first n terms of the series formed by squaring the natural numbers.

>   Sample output: <br>
>   Enter a number: 6 <br>
>   The first 6 terms of the series are:
>   1 4 99 16 25 36

In [6]:
num = int(input("Enter a number: "))

print(f"The first {num} terms of the series are:")

for i in range(1, num + 1):
    print(i * i, end=" ")


Enter a number:  15


The first 15 terms of the series are:
1 4 9 16 25 36 49 64 81 100 121 144 169 196 225 

<h4 style="color:green;">Q8. Write a program that prompts the user to input a number and prints its multiplication table.

In [7]:
num = int(input("Enter a number: "))

print(f"\nMultiplication Table of {num}:")
for i in range(1, 11):
    print(f"{num} x {i} = {num * i}")


Enter a number:  15



Multiplication Table of 15:
15 x 1 = 15
15 x 2 = 30
15 x 3 = 45
15 x 4 = 60
15 x 5 = 75
15 x 6 = 90
15 x 7 = 105
15 x 8 = 120
15 x 9 = 135
15 x 10 = 150


<h4 style="color:green;">Q9. Write a Python program to print the first 8 terms of an arithmetic progression starting with 3 and having a common difference of 4. <br>

>   The program should output the following sequence: <br>
>   3 7 11 15 18 23 27 31 <br>  

In [8]:
a = 3      #--> first term
d = 4      #--> common difference
n = 8      #--> number of terms

print("The first 8 terms of the AP are:")

for i in range(n):
    term = a + i * d                     #--> formula arithmetic progression
    print(term, end=" ")

The first 8 terms of the AP are:
3 7 11 15 19 23 27 31 

<h4 style="color:green;">Q10. Write a Python program to print the first 6 terms of a geometric sequence starting with 2 and having a common ratio of 3. <br>
The program should output the following sequence: <br>

>   The program should output the following sequence: <br>
>   2 6 18 54 162 486

In [9]:
a = 2      #--> first term
r = 3      #--> common ratio
n = 6      #--> number of terms

print("The first 6 terms of the GP are:")

for i in range(n):
    term = a * (r ** i)                  #--> formula geometric sequence
    print(term, end=" ")

The first 6 terms of the GP are:
2 6 18 54 162 486 

<h4 style="color:green;">Q11. Write a program that asks the user for a positive integer value. The program should calculate the sum of all the integers from 1 upto the number entered. For examples, if the user enters 20 the loop will find the sum of 1, 2, 3, 4, ... 20.

In [10]:

num = int(input("Enter a positive integer: "))

if num > 0:
    total = 0
    for i in range(1, num + 1):
        total += i
    print(f"The sum of numbers from 1 to {num} is: {total}")
else:
    print("Please enter a positive integer.")

Enter a positive integer:  15


The sum of numbers from 1 to 15 is: 120


<h4 style="color:green;">Q12. Write a program that takes a positive integer N as input and calclulates the sum of the reciprocals of all numbers from 1 up to N. The program should display the final sum.
    
>   Output of the program should be like: <br>
>   Enter a positive integer: 5 <br>
    The sum of reciprocals from 1 to 5 is: 2.28

In [12]:
N = int(input("Enter a positive integer: "))

if N > 0:
    total = 0
    for i in range(1, N + 1):
        total += 1 / i
    print(f"The sum of reciprocals from 1 to {N} is: {total:.2f}")
else:
    print("Please enter a positive integer.")

Enter a positive integer:  5


The sum of reciprocals from 1 to 5 is: 2.28


<h4 style="color:green;">Q13. Write a program that prompts the user to enter a number and repeat this process 5 times. The program should accumulate the numbers entered and then display the final running total. <br>

>   Sample Output: <br>
    Enter a number: 10 <br>
    Enter a number: 15 <br>
    Enter a number: 35 <br>
    Enter a number: 40 <br>
    Enter a number: 50 <br>
    The final running total is: 150 


In [13]:
total = 0

for i in range(5):
    num = int(input("Enter a number: "))
    total += num

print(f"The final running total is: {total}")

Enter a number:  10
Enter a number:  15
Enter a number:  35
Enter a number:  40
Enter a number:  50


The final running total is: 150


<h4 style="color:green;">Q14.	Write a program that prompts the user to enter a positive integer and calculates its factorial. The factorial of a positive integer 'n' is denoted as 'n!' and is calculated by multiplying all the integers from 1 to 'n' together. For example, the factorial of 5 (denoted as 5!) is calculated as 1 x 2 x 3 x 4 x 5.
    
The program should display the factorial value if the input is a positive number, or display a message stating that the factorial does not exist for negative numbers.
Additionally, for an input of zero, the program should output that the factorial of 0 is 1.

In [14]:
num = int(input("Enter a number: "))

if num < 0:
    print("Factorial does not exist for negative numbers.")
elif num == 0:
    print("The factorial of 0 is 1")
else:
    factorial = 1
    for i in range(1, num + 1):
        factorial *= i
    print(f"The factorial of {num} is {factorial}")

Enter a number:  5


The factorial of 5 is 120


<h4 style="color:green;">Q15. Write a Python program that prompts the user to enter a base number and an exponent, and then calculates the power of the base to the exponent. The program should not use the exponentiation operator (**) or the math.pow() function. The program should handle both positive and negative exponents.

In [17]:
base = float(input("Enter the base number: "))
exp = int(input("Enter the exponent: "))

result = 1

if exp > 0:
    for i in range(exp):
        result *= base
elif exp < 0:
    for j in range(-exp):
        result *= base
    result = 1 / result

print(f"{base} raised to the power {exp} is: {result}")

Enter the base number:  6
Enter the exponent:  3


6.0 raised to the power 3 is: 216.0
